### PR TITLE
[MathCore][ROOT-10059] Set/Get state of TRandom3

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

**Comment by amadio** [comment]
@phsft-bot build just on windows10/default

File: math/mathcore/inc/TRandom3.h [hunk 1]
```diff
@@ -24,6 +24,19 @@
...
+class TRandom3;
+
+class TRandom3State {
+   friend TRandom3;
+   std::array<UInt_t,624> fMt;
+   Int_t fCount624;
+   TRandom3State(const UInt_t *mt, Int_t count624) : fCount624(count624) {
+      std::copy_n(mt, 624, fMt.begin());
+   }
+};
```
Comment (amadio) [inline math/mathcore/inc/TRandom3.h]: How about putting this class inside TRandom3 itself? That way the details of the state stay opaque and can be changed later if needed.

Comment (dpiparo) [reply math/mathcore/inc/TRandom3.h]: what do you think @lmoneta ?

**Comment by lmoneta** [comment]
What is the use case for this PR ? 
I am not sure it is a good idea to expose the internal RNG state. It is well known that Mersenne-Twister has several defects, in particular to suffer from states where several zeros are present. 
This PR will allow users to create states which can produce bogus random numbers. 

Lorenzo

**Comment by lmoneta** [comment]
I see now the Jira item ROOT-10059. The correct solution for this is to disable GetSeed() for TRandom3. If one wants to copy the state it is just enough to use the class copy constructor

**Comment by dpiparo** [comment]
cool. I close this PR. Can you close the related jira item?

**Pull request closed without merging**